code/modules/clothing/head/helmet.dm

```diff
@@ -11,6 +11,22 @@
 	heat_protection = HEAD
 	max_heat_protection_temperature = HELMET_MAX_TEMP_PROTECT
 	strip_delay = 60
+	can_flashlight = 1
+	var/obj/machinery/camera/helmetCam = null
+	var/spawnWithHelmetCam = 0
+	var/canAttachCam = 1
+
+
+/obj/item/clothing/head/helmet/New()
+	..()
+	if(spawnWithHelmetCam)
+		helmetCam = new /obj/machinery/camera(src)
+		helmetCam.c_tag = "Helmet-Mounted Camera (real_name)([rand(1,999)])"

```

Comment:
this "real_name" should be "No User" again, as I had originally.
The equip_to_slot_or_del of the officer should call equipped() already meaning the user info will get updated fine.